modules/core/src/main/java/org/apache/ignite/internal/jdbc/thin/JdbcThinConnection.java

```diff
@@ -223,6 +261,10 @@
     public JdbcThinConnection(ConnectionProperties connProps) throws SQLException {
         this.connProps = connProps;
 
+        this.metaHnd = new JdbcBinaryMetadataHandler();
```

Comment:
I think it is not good idea, because we don't know which connections should have common registry and which not. As a result, we may get collisions by _className_ or _typeId_.